### DISCUSSION TITLE
Styling semantic tokens with CSS and custom fonts?

### DISCUSSION THREAD

**Discussion by phlegmaticprogrammer** [microsoft/vscode-discussions#381] [Extension Development QnA] [answered]
I am writing an extension for a language that needs its own font to display nicely, so I would like to ship it with the extension. Is there a good way to do this? Any pointers much appreciated. Currently I am using a configuration default of 
```
"editor.fontFamily": "STIXGeneral"
```
but that assumes that the STIXGeneral font is already installed on the users machine.

Furthermore, I would like to be able to switch fonts (and other css attributes) depending on the semantic token. Currently only limited options like setting the color, bold and italic are available. Is there a way to do this? Again, pointers much appreciated!

Best,

Steven Obua

**Comment by isidorn** [comment, marked as answer]
I think shipping the font with your extension is the way to go. Not sure how large the fonts are?

**Reply by phlegmaticprogrammer** [reply]
Yes, I definitely would like to ship the font with the extension. Size is not really an issue. But what I am wondering about is the following:

Is there a mechanism for VSCode to install a font I ship with my extension automatically? I don't see any contribution point for font files documented.

**Reply by isidorn** [reply]
No, there is no such contribution point. Your extension would have to do it on it's own.
